src/helpers.php

```diff
@@ -1,15 +1,32 @@
 <?php
 
-use MikeFrancis\LaravelSecureHeaders\CSPNonce;
+use Illuminate\Container\Container;
+use MikeFrancis\LaravelSecureHeaders\Exceptions\ContentSecurityPolicyNotFoundException;
 
 if (!function_exists('csp_nonce')) {
     /**
-     * Provide a base-64 CSP nonce that can be used for inline styles.
-     * The nonce will be regenerated once per pageload.
+     * The nonce will be regenerated once per page load.
+     *
+     * @param string $friendlyDirective
      * @return string
+     * @throws ContentSecurityPolicyNotFoundException
      */
-    function csp_nonce(): string
+    function csp_nonce(string $friendlyDirective)
     {
-        return CSPNonce::get();
+        $app = Container::getInstance();
+        $request = $app['request'];
+        $cspHeader = $request->headers->get('content-security-policy');
+
+        if (!$cspHeader) {
+            throw new ContentSecurityPolicyNotFoundException();
+        }
+
+        $parts = explode(' ', $cspHeader, 2);
+
+        if ($parts[0] === $friendlyDirective) {
```

Comment:
I think you'll probably need to parse the CSP header here, this method would only work in a few special cases (and even then the "nonce" that this would return would need to be modified to get the nonce value –
 for example, if the nonce is `1234` it would look like `'nonce-1234'` as a source item in the CSP header, but the user needs to use the nonce value in their scripts). 

The structure of a CSP header roughly corresponds to an array of the form `array('directive-1' => array('source-1', 'source-2', ...), ...)` where the nonce is one of the source items.

Each directive has a source list, and these coupled together are delimited with `;`. Once you have a directive coupled with source list then you can split on the first space to separate the directive from its source values, and then get the source values for that directive by splitting on the spaces thereafter.

The exact syntax is specified here: https://www.w3.org/TR/CSP2/#policy-syntax
You also might like to take a look at [`SecureHeaders\Headers\CSPHeader`](https://github.com/aidantwoods/SecureHeaders/blob/9d5160170971336af85cbff9a6d721f42575ef1c/src/Headers/CSPHeader.php) and [`SecureHeaders\Operations\CompileCSP::deconstructCSP`](https://github.com/aidantwoods/SecureHeaders/blob/9d5160170971336af85cbff9a6d721f42575ef1c/src/Operations/CompileCSP.php#L157-L163), but this isn't part of the [public API](https://github.com/aidantwoods/SecureHeaders#versioning), so using this method directly isn't guaranteed to work across releases (in-fact I'm planning some restructuring so I can pretty much guarantee this *will* break at some point if used directly).

---

If its possible to access the `SecureHeaders` instance that will have been used to on the headers, you can just directly pull a nonce out of its internal storage, without having to worry about parsing headers, with https://github.com/aidantwoods/SecureHeaders/wiki/cspNonce. But be aware that this is a mutating method that will create a nonce if one doesn't exist. So you'd either want to ensure that `apply` is called at some point after (so that the new nonce is actually added to the headers that get sent), or just lookup the directive being requested against your config to see if one would have been added. And then if it appeared in the config, ask SecureHeaders for its value. The latter is really the only correct solution, in the case that output has already started to be sent (since the headers are immutable by PHP at this point).

I'm assuming that SecureHeaders will have already been applied to the response headers by the time the user can call this method?

Reply:
> I'm assuming that SecureHeaders will have already been applied to the response headers by the time the user can call this method?

Yes - this method I imagine will be called by a templating engine so at that point the headers have been applied so all we have to get them is grabbing the request out of the container and trying to find the header.